package/MDAnalysis/analysis/MSD.py

```diff
@@ -0,0 +1,398 @@
+    >>>
+    >>> my_universe = MDAnalysis.Universe(waterPSF, waterDCD)
+
+Next, create an MSD object and run the calculation:
+
+    >>> my_MSD = MSD(my_universe, ['type OT','type HT'],0,9,5)
+    >>> my_MSD.run()
+
+This line initializes an MSD object by passing it the universe object, a list
+of selection groups to calculate, the first and last frame numbers to analyze,
+and the number of frames between each restart point in the MSD calculation.
+Note that this script is capable of calculating the MSD of any atom selection,
+regardless of complexity, so the MSD of something like
+ 'type OT around 4 type HT' can be calculated.
+
+By default, the MSD object will write output in binary to a pickle file named
+'msd' which will contain MSD calculated for tau less than 250 frames or the
+length of the trajectory, which ever is smaller. Data in this file is in the
+form: [msd,n_samples], where msd is a vector of length=2 in this example,
+containing vectors of MSD values with indexes corresponding to tau,
+and n_samples is a similar array containing the number of distances used
+to calculate the msd at the associated index.
+
+To change these defaults, you can use the optional parameters 'out_name' and
+'len_msd':
+
+    >>> my_MSD = MSD(my_universe, ['type OT','type HT'],0,9,5,
+                     out_name='my_new_name.p',len_msd=5)
+
+Setting write_output=False will cause my_MSD.run() to return the data it would
+otherwise write to the output file. Finally, setting max_data=False will allow
+the calculation to use data past the specified last frame in the msd
+calculation. This can be useful for splitting large jobs over multiple batch
+scripts, as it ensures that no data is lost in the process.
+
+"""
+
+
+from __future__ import absolute_import, print_function, division
+from six import range
+
+import sys
+import getopt
+import warnings
+from collections import deque
+
+import numpy as np
+import pickle
+
+import MDAnalysis
+from MDAnalysis.lib.distances import squared_distance_vector
+
+
+class MSD(object):
+    """Object for calculating mean square displacement (MSD)
+
+    Attributes
+    ----------
+    pos : list
+        contains position information, organized:
+        selection X frame X atom X (x, y, z)
+    atomids_per_sel_at_t : list
+        contains set of atoms which satisfy each selection at each
+        frame, organized:
+        selection X frame X set of atom ids
+    map_atomids_to_pos_at_t : list
+        contains dictionaries which map atom ids to corresponding
+        pos index, organized:
+        selection X frame X dict
+    msd : np.array(dtype=float64)
+        Contains the sum of squared displacements data, organized:
+        selection X tau
+        (MSD is calculated as msd/n_samples)
+    n_samples : np.array(dtype=int)
+        Contains the number of samples corresponding to each msd
+        entry, organized:
+        selection X tau
+
+    Methods
+    -------
+    _select(self)
```

Comment:
You don't need to document private methods. Only the public methods to interact with the class.